### PR TITLE
Event registration changes

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Form/EventRegistrationType.php [hunk 1]
```diff
@@ -9,21 +9,41 @@
...
-                'required' => false,
+            ->add('acceptTerms', CheckboxType::class, [
+                'mapped' => false,
             ])
         ;
+
+        /** @var EventRegistrationCommand $registration */
+        $registration = $builder->getData();
+        if ($builder->getData() instanceof EventRegistrationCommand
+            && (!$this->authorizationChecker->isGranted('IS_AUTHENTICATED_FULLY')
```
Comment (HeahDude) [inline src/Form/EventRegistrationType.php]:
```
If (!$this->authorizationChecker(...) || ($resgistration instanceof ... && !$registration->is...)) {
```

Comment (illusionOfParadise) [reply src/Form/EventRegistrationType.php]: Changed in this PR https://github.com/EnMarche/en-marche.fr/pull/2024/files#diff-44eb87946c3878749aba2837e96ea962R38

File: tests/Controller/CitizenActionControllerTest.php [hunk 2]
```diff
@@ -108,7 +168,18 @@ public function testUnregistrationFailed(): void
         $this->client->request(Request::METHOD_GET, sprintf('/action-citoyenne/%s', $citizenAction->getSlug()));
 
         $this->assertSame('S\'inscrire', $this->client->getCrawler()->filter('a.newbtn--orange')->text());
-        $this->assertNull($this->getEventRegistrationRepository()->findAdherentRegistration($uuid, LoadAdherentData::ADHERENT_2_UUID));
+        $this->assertNull($this->repository->findAdherentRegistration($uuid, LoadAdherentData::ADHERENT_2_UUID));
+    }
+
+    private function seeMessageSuccesfullyCreatedFlash(Crawler $crawler, ?string $message = null)
```
Comment (HeahDude) [inline tests/Controller/CitizenActionControllerTest.php]: It's not the first time I see this method in a test we should move it to the `ControllerTestTrait` and rename it `seeFlashMessage`.
Also, when no message is passed, we should assert that `0 === count($flash)`.

File: src/DataFixtures/ORM/LoadCitizenProjectCommentData.php [hunk 3]
```diff
@@ -16,14 +16,16 @@ public function load(ObjectManager $manager)
         $comment1 = $this->createCitizenProjectComment(
             $this->getReference('citizen-project-1'),
             $this->getReference('adherent-2'),
-            'Jean-Paul à Maurice : tout va bien ! Je répète ! Tout va bien !'
+            'Jean-Paul à Maurice : tout va bien ! Je répète ! Tout va bien !',
+            '-1 minute'
```
Comment (HeahDude) [inline src/DataFixtures/ORM/LoadCitizenProjectCommentData.php]: Yay on this, it will prevent our "random" failing test, caused by a race condition IMO.

Comment (illusionOfParadise) [reply src/DataFixtures/ORM/LoadCitizenProjectCommentData.php]: Yes, I added this to fix this "random" failing.

File: tests/Controller/ControllerTestTrait.php [hunk 4]
```diff
@@ -74,6 +75,17 @@ public function authenticateAsAdherent(Client $client, string $emailAddress, str
         return $client->followRedirect();
     }
 
+    public function seeFlashMessage(Crawler $crawler, ?string $message = null): bool
```
Comment (HeahDude) [inline tests/Controller/ControllerTestTrait.php]: Can remain protected.

**Pull request merged**